src/ExceptionConverterHelper.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App;
+
+use Keboola\DockerBundle\Docker\Runner\Output;
+use Keboola\DockerBundle\Exception\UserException;
+use Keboola\ErrorControl\Message\ExceptionTransformer;
+use Keboola\JobQueueInternalClient\JobFactory\JobResult;
+use Keboola\ObjectEncryptor\Exception\UserException as EncryptionUserException;
+use Psr\Log\LoggerInterface;
+use Throwable;
+
+class ExceptionConverterHelper
+{
+    public static function convertExceptionToResult(
+        LoggerInterface $logger,
+        Throwable $e,
+        string $jobId,
+        array $outputs
+    ): JobResult {
+        $errorType = is_a($e, UserException::class) || is_a($e, EncryptionUserException::class)
+            ? JobResult::ERROR_TYPE_USER : JobResult::ERROR_TYPE_APPLICATION;
+        if (is_a($e, UserException::class)) {
+            $errorTypeString = 'user';
+        } elseif (is_a($e, EncryptionUserException::class)) {
+            $errorTypeString = 'encryption';
+        } else {
+            $errorTypeString = 'application';
+        }
```

Comment:
Osobne bych ten `$errorType` resolvoval v ramci toho `if/elseif/else` bloku. Mas tam sice 2x ten user type, ale prijde mi to prehlednejsi, protoze premyslis jen nad jednim blokem podminek.

```suggestion
        if (is_a($e, UserException::class)) {
            $errorTypeString = 'user';
            $errorType = JobResult::ERROR_TYPE_USER;
        } elseif (is_a($e, EncryptionUserException::class)) {
            $errorTypeString = 'encryption';
            $errorType = JobResult::ERROR_TYPE_USER;
        } else {
            $errorTypeString = 'application';
            $errorType = JobResult::ERROR_TYPE_APPLICATION;
        }
```